### PR TITLE
Fix linting on Xcode 8

### PR DESCRIPTION
**Pull request by DanToml**
Will default to SWIFT_VERSION = 2.3, but will check for a `.swift-version` file and use the version specified there if present.

### CODE REVIEW

File: lib/cocoapods/validator.rb [hunk 1]
```diff
@@ -237,6 +237,19 @@ def validation_dir
...
+    #
+    def swift_version
+      @swift_version ||= dot_swift_version || '2.3'
+    end
+
+    # @return [String] the SWIFT_VERSION in the .swift-version file or nil.
+    #
+    def dot_swift_version
+      swift_version_path = file.dirname + '.swift-version'
+      File.read(swift_version_path) if File.exist?(swift_version_path)
```
Comment (segiddins) [inline lib/cocoapods/validator.rb]: let's use Pathname methods here?

**Comment by segiddins** [comment]
👍 other than the one nit

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! 
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* Fix linting on Xcode 8 
  [DanToml](https://github.com/DanToml)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Comment by AnthonyMDev** [comment]
What is the syntax for the contents of the `.swift-version` file? I can't seem to find documentation of this anywhere yet.

**Comment by orta** [comment]
They're just the version number - for example: https://github.com/kylef/Stencil/blob/master/.swift-version

@benasher44 we should note this in the blog post

**Comment by AnthonyMDev** [comment]
Excellent. Thanks.

**Comment by xmkevinchen** [comment]
Hey, I still got error  for command`pod lib lint`

Here's my `pod env`

### Stack

```
   CocoaPods : 1.1.0.rc.1
        Ruby : ruby 2.2.1p85 (2015-02-26 revision 49769) [x86_64-darwin14]
    RubyGems : 2.4.8
        Host : Mac OS X 10.11.6 (15G1004)
       Xcode : 8.0 (8A218a)
         Git : git version 2.10.0
Ruby lib dir : /chen_kevin.rvm/rubies/ruby-2.2.1/lib
Repositories : indooratlas - https://github.com/IndoorAtlas/CocoaPods-Specs @ dd2c6824b23d6cb578a0ce542574173796bc154f
               master - https://github.com/CocoaPods/Specs.git @ 7388fbb06d9848dc512183fb5e76f5a66985b97c
```

### Installation Source

```
Executable Path: /chen_kevin.rvm/gems/ruby-2.2.1/bin/pod
```

### Plugins

```
cocoapods-clean       : 0.0.1
cocoapods-deintegrate : 1.0.1
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.0.0
cocoapods-try         : 1.1.0
```

Here's my .swift-version file 

```
cat .swift-version
3.0
```

**Comment by benasher44** [comment]
Hi @xmkevinchen! This will be fixed in .rc.2 (next release soon).

**Comment by humblehacker** [comment]
Looking forward to this change.  Out of curiosity, why a dot file instead of an option in Podspec?

**Comment by pip8786** [comment]
Hey @benasher44 What is the time frame on "soon"? Hours, days, weeks? Just wondering so I can plan my updates accordingly. Thanks.

**Comment by benasher44** [comment]
Day to days

**Comment by pip8786** [comment]
Alright thanks. I'll hold off on trying to update to swift 3 until then.

**Comment by DanToml** [comment]
@pip8786 I released 1.1.0.rc.2 today

**Comment by pip8786** [comment]
@DanToml Thanks! Things seem to be working now with a Swift 3 pod.

**Comment by Coeur** [comment]
It should have been in the podspec instead of a new file

**Comment by freak4pc** [comment]
I would also be a fan of the `Podfile` configuration instead of a separate file, but any solution is a good one, practically :) Great job @DanToml 👍

**Comment by remirobert** [comment]
same issue 😢

**Comment by jonasman** [comment]
Why not defaulting to 3.0 ?

**Comment by DanToml** [comment]
Hi all thanks for the feedback, I'm locking this PR now. If there are any specific bugs with this please file a new issue and fill in the Issue Template.

**Pull request merged**